src/services/AsyncStorage.ts

```diff
@@ -76,6 +80,35 @@ export const saveExerciseProgress = async (
   await setExerciseProgress(disciplineId, exerciseKey, score)
 }
 
+const getFavorites = async (): Promise<Document[]> => {
+  const documents = await AsyncStorage.getItem(FAVORITES_KEY)
+  return documents ? JSON.parse(documents) : []
+}
+
+const setFavorites = async (favorites: Document[]): Promise<void> => {
```

Comment:
I think it would make more sense to only save id's of the favorites, and request the other data from the api, as images improve, alternative words are added and so on.

Reply:
Blocked by https://issues.tuerantuer.org/browse/LUN-307 (or rather: https://github.com/digitalfabrik/lunes-cms/issues/324).

Reply:
hmh, we can not deploy your version and switch later, as users may have already saved whole words as favorites. 

So either we have to disable this feature or we use one of the existing endpoints and optimize this, when the cms provides the new endpoint. I prefer the second.

Reply:
I'm also with steffi even if it gonna be quite slow at the moment (2,5s), i would prefer to save ids and maybe fetch all the words and filter in the frontend. Lateron we just pass these wordIds to the new endpoint

Reply:
done